### PR TITLE
Document 'securing key value store'

### PR DESCRIPTION
**Pull request by lakshmi-kannan**

### CODE REVIEW

File: docs/source/datastore.rst [hunk 1]
```diff
@@ -130,3 +130,80 @@ related syntax.
         }
     }
 
+Securing secrets in key value store (admin only)
+------------------------------------------------
+
+.. note::
+
+    This guide and the corresponding implementation is alpha quality. We are working on the feature
+    and feedback is welcome. Unless the feature matures and deployment issues identified and fixed,
```
Comment (lakshmi-kannan) [inline docs/source/datastore.rst]: Note to self: s/Unless/Until. LOL

File: docs/source/datastore.rst [hunk 2]
```diff
@@ -130,3 +130,80 @@ related syntax.
...
+    no guarantee is made w.r.t ``security`` of the credentials stored in key value store.
```
Comment (manasdk) [inline docs/source/datastore.rst]: I love it when we come with a security feature and then say beware dragon be living here :P

File: docs/source/datastore.rst [hunk 3]
```diff
@@ -130,3 +130,80 @@ related syntax.
...
+
+Key value store now allows users to store encrypted credentials via a global symmetric key set
+by the StackStorm admin. It goes without saying that admin can decrypt the credentials if they
+wants to.
```
Comment (manasdk) [inline docs/source/datastore.rst]: s/wants/want

File: docs/source/datastore.rst [hunk 4]
```diff
@@ -130,3 +130,80 @@ related syntax.
...
+To generate a symmetric crypto key (AES256 used for both encryption and decryption) as an admin,
+please run
+
+.. code-block:: bash
+
+    /opt/stackstorm/st2/tools/st2-generate-symmetric-crypto-key.py --key-path /path/to/key/file.json
+
+It is recommended that the key is placed in a private location such as /etc/st2/keys/ and
+permissions are appropriately modified so only StackStorm API process owner (usually ``st2``) and
+admin can read/write to that file.
```
Comment (manasdk) [inline docs/source/datastore.rst]: API only need read access correct?

Comment (lakshmi-kannan) [inline docs/source/datastore.rst]: Good point. Will make it clear.

File: docs/source/datastore.rst [hunk 5]
```diff
@@ -130,3 +130,80 @@ related syntax.
...
+Security notes
+--------------
+
+|st2| wishes to discuss security details and be transparent about the implementation and limitations
+of the security practices to attract more eyes to it and therefore build better quality into
+security implementations. For the key value store, we have settled on AES256 symmetric encryption
+for simplicity. We use python library keyczar for doing this.
+
+We have made a trade off that |st2| admin is allowed to decrypt the secrets in key value store.
+This made our implementation simpler. We are looking into how to let users pass their own keys
```
Comment (manasdk) [inline docs/source/datastore.rst]: "This made out implementation simpler" sounds along the lines of we took the easy way out :)

Comment (lakshmi-kannan) [inline docs/source/datastore.rst]: Honesty.

**Comment by manasdk** [comment]
WFM

**Pull request merged**